llama_index/vector_stores/supabase.py

```diff
@@ -0,0 +1,133 @@
+import math
+from typing import Any, List
+
+from llama_index.data_structs.node import DocumentRelationship, Node
+from llama_index.vector_stores.types import (
+    MetadataFilters,
+    NodeWithEmbedding,
+    VectorStore,
+    VectorStoreQuery,
+    VectorStoreQueryResult,
+)
+
+
+class SupabaseVectorStore(VectorStore):
+    """Supbabase vector store.
+
+    In this vector store, embeddings are stored within a pg-vector collection.
+
+    During query time, the index uses pg-vector/supabase to query for the top
+    k most similar nodes.
+
+    Args:
+        postgres_connection_string (str):
+            postgres connection string
+
+        collection_name (str):
+            name of the collection to store the embeddings in
+
+    """
+
+    stores_text = True
+
+    def __init__(
+        self, postgres_connection_string: str, collection_name: str, **kwargs: Any
+    ) -> None:
+        """Init params."""
+        import_err_msg = "`vecs` package not found, please run `pip install vecs`"
+        try:
+            import vecs
+        except ImportError:
+            raise ImportError(import_err_msg)
+
+        client = vecs.create_client(postgres_connection_string)
+
+        self._collection = client.get_collection(name=collection_name)
+
+    @property
+    def client(self) -> None:
+        """Get client."""
+        return None
+
+    def _to_vecs_filters(self, filters: MetadataFilters) -> Any:
+        """Convert llama filters to vecs filters. $eq is the only supported operator."""
+        vecs_filter = {}
+        for f in filters:
+            vecs_filter[f[0]] = {"$eq": f[1]}
+        return vecs_filter
+
+    def add(self, embedding_results: List[NodeWithEmbedding]) -> List[str]:
+        """Add embedding results to index.
+
+        Args
+            embedding_results: List[NodeWithEmbedding]: list of embedding results
+
+        """
+        if not self._collection:
+            raise ValueError("Collection not initialized")
+
+        data = []
+        ids = []
+        for result in embedding_results:
+            data.append((result.id, result.embedding, result.node.text))
```

Comment:
nit: would be great to add `Node` metadata in addition to just the text here, so that we can apply the metadata filters properly.

Reply:
I think here we can reuse the `node_to_metadata_dict(result.node))` method (from `llama_index.vector_stores.utils`) as seen in the Chroma/Pinecone impl.

Reply:
Great suggestion. Pushed a new version - lmk if that's what you had in mind?

Reply:
Taking a look!